src/main/java/com/boot/amazon/controller/ReviewController.java

```diff
@@ -14,15 +25,43 @@
 public class ReviewController {
     private final ReviewService reviewService;
     private final ReviewMapper reviewMapper;
+    private final UserService userService;
+    private final ProductSevice productSevice;
 
-    public ReviewController(ReviewService reviewService, ReviewMapper reviewMapper) {
+    public ReviewController(ReviewService reviewService,
+                            ReviewMapper reviewMapper, UserService userService,
+                            ProductSevice productSevice) {
         this.reviewService = reviewService;
         this.reviewMapper = reviewMapper;
+        this.userService = userService;
+        this.productSevice = productSevice;
     }
 
-    @GetMapping("/top-words")
+    @GetMapping("/admin/top-words")
     public ReviewRequestDto getWords(@RequestParam(defaultValue = "1000") int limit) {
         List<String> words = reviewService.getTopMostFrequentWordsInReview(limit);
         return reviewMapper.map(words);
     }
+
+    @DeleteMapping("/admin/delete")
+    public void deleteReview(@RequestParam Long reviewId) {
+        reviewService.deleteById(reviewId);
+    }
+
+    @PostMapping("/add")
+    public void addReview(@RequestBody ReviewRequestAddDto requestAddDto) {
+        User user = userService.findByProfileName(requestAddDto.getProfileName()).orElseThrow();
+        Product product = productSevice
+                .findByproductIdFromSource(requestAddDto.getProductId()).orElseThrow();
+        reviewService.save(reviewMapper.map(requestAddDto, user, product));
+    }
+
+    @PostMapping("/change")
```

Comment:
request to change some resources should be indempotent( sending a few times the same result).So it's better use PUT method for that purpose